tests/integrationv2/test_buffered_send.py

```diff
@@ -0,0 +1,166 @@
+
+K_BYTES = 1024
+SEND_BUFFER_SIZE_MIN = 1031
+SEND_BUFFER_SIZE_MIN_RECOMMENDED = 2 * K_BYTES
+SEND_BUFFER_SIZE_MULTI_RECORD = 17 * K_BYTES
+SEND_BUFFER_SIZE_PREFER_THROUGHPUT = 35 * K_BYTES
+SEND_BUFFER_SIZE_HUGE = 512 * K_BYTES
+
+SEND_BUFFER_SIZES = [
+    SEND_BUFFER_SIZE_MIN,
+    SEND_BUFFER_SIZE_MIN_RECOMMENDED,
+    SEND_BUFFER_SIZE_MULTI_RECORD,
+    SEND_BUFFER_SIZE_PREFER_THROUGHPUT,
+    SEND_BUFFER_SIZE_HUGE
+]
+
+FRAGMENT_PREFERENCE = [
+    None,
+    "--prefer-low-latency",
+    "--prefer-throughput"
+]
+
+
+def test_SEND_BUFFER_SIZE_MIN_is_s2ns_min_buffer_size(managed_process):
+    port = next(available_ports)
+
+    s2n_options = ProviderOptions(mode=Provider.ServerMode,
+        port=port,
+        data_to_send="test",
+        extra_flags=['--buffered-send', SEND_BUFFER_SIZE_MIN-1])
+
+    s2nd = managed_process(S2N, s2n_options)
+
+    s2n_options.mode = Provider.ClientMode
+
+    s2nc = managed_process(S2N, s2n_options)
+
+    for results in s2nd.get_results():
+        assert "Error setting send buffer size" in str(results.stderr)
+        assert results.exit_code != 0
+
+    for results in s2nc.get_results():
+        assert "Error setting send buffer size" in str(results.stderr)
+        assert results.exit_code != 0
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("peer", [GnuTLS, OpenSSL, S2N], ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", PROTOCOLS, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", MINIMAL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("buffer_size", SEND_BUFFER_SIZES, ids=get_parameter_name)
+@pytest.mark.parametrize("fragment_preference", FRAGMENT_PREFERENCE, ids=get_parameter_name)
+def test_s2n_server_buffered_send(managed_process, cipher, peer, protocol, certificate, buffer_size, fragment_preference):
+    # Communication Timeline
+    # Client [S2N|OpenSSL|GnuTLS]  | Server [S2N]
+    # Handshake                    | Handshake
+    #                              | Send SEND_DATA (with CLOSE_MARKER)
+    # Closes with CLOSE_MARKER     | Close
+    port = next(available_ports)
+
+    peer_client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        port=port,
+        cipher=cipher,
+        data_to_send=None,
+        insecure=True,
+        protocol=protocol,
+        verbose=False)
+
+    extra_flags = ['--buffered-send', buffer_size]
+    if fragment_preference is not None:
+            extra_flags.append(fragment_preference)
+
+    s2n_server_options = ProviderOptions(
+        mode=Provider.ServerMode,
+        port=port,
+        cipher=cipher,
+        data_to_send=SEND_DATA,
+        insecure=True,
+        protocol=protocol,
+        key=certificate.key,
+        cert=certificate.cert,
+        extra_flags=extra_flags)
+
+    server = managed_process(S2N, s2n_server_options, send_marker=[S2N.get_send_marker()])
```

Comment:
send_marker doesn't default to the provider's send_marker?

Reply:
Not on the managed_process for servers.

https://github.com/harrisonkaiser/s2n-tls/blob/b91b9bcc670922e863fbc3d10d413b8a7265898e/tests/integrationv2/providers.py#L35